### PR TITLE
Move to bulk queries where relevant [ClientsPresenter#group_projects_by_client_name]

### PR DESCRIPTION
**Pull request by mayankagnihotri7**
## Notion card
[Notion](https://www.notion.so/saeloun/Move-to-bulk-queries-where-relevant-210108ab54674c78b3f5e948965e3a7d)

## Summary

Refactor the query for getting projects for the employee.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking and retains same functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [ ] I have annotated changes in the PR that are relevant to the reviewer
- [ ] I have added automated tests for my code

### CODE REVIEW

**Comment by supriya3105** [comment]
@mayankagnihotri7 Can you finish the review comments on this PR and check the failing tests when you have the time from client work.

**Comment by mayankagnihotri7** [comment]
@swaathi I've made the requested changes. Please review.

**Comment by supriya3105** [comment]
@mayankagnihotri7 please check why the rspec tests are failing on this PR . The status of the PR should be green.

File: app/presenters/clients_presenter.rb [hunk 1]
```diff
@@ -14,11 +14,9 @@ def group_projects_by_client_name
     if is_admin
       clients.each { |client| projects[client.name] = client.projects.kept }
     else
-      employee_projects = current_user.projects.kept.joins(:client).where(clients: { company_id: current_company.id })
-      client_projects = clients.joins(:projects).select(
+      current_user.projects.kept.joins(:client).where(clients: { company_id: current_company.id }).select(
```
[the remark this answers is not in the document]
Comment (supriya3105) [reply app/presenters/clients_presenter.rb:18]: @mayankagnihotri7 Please respond to above review comment from Swaathi. Also, the failing tests. If you are working on it , please add the findings and progress here.

**Pull request closed without merging**